### ISSUE TITLE
LTA error with regex in `where` clause while `$/` is used for a later param in the same signature

### ISSUE THREAD

**Issue by vrurg** [rakudo/rakudo#2059]
## The Problem

When a grammar is using a `proto` with a token or a rule and is combined with an action class with `FALLBACK()` method defined then if a regex constraint applied to the first `FALLBACK()` argument then rakudo generates a exception.

## Expected Behavior

`FALLBACK()` must be executed if constraint is fullfilled.

## Actual Behavior

The following exception is raised:

```
Cannot call method 'STORE' on a null object
```

## Steps to Reproduce

The following script demonstrates the behavior:

```
grammar mdHTMLEntity {
    rule TOP { <mdHTMLEntity>+ }

    token mdHTMLEntity { '&' <mdHTMLCharReference> ';' }

    proto token mdHTMLCharReference {*}
    token mdHTMLCharReference:sym<dec> { [ '#' <[0..9]>+ ] }
    token mdHTMLCharReference:sym<hex> { [ :i '#x' <[0..9a..f]>+ ] }
    token mdHTMLCharReference:sym<named> { [ amp | lt | gt ] }
}

class A {
    multi method FALLBACK ($name where /^md/, $/) {
        say "fallback for $name";
    }
}

my $res = mdHTMLEntity.parse( "&#88;", actions => A.new );
say $res;
```

## Environment
* Operating system: macOS 10.13.5
* Compiler version (`perl6 -v`): This is Rakudo version 2018.06 built on MoarVM version 2018.06

**Comment by zoffixznet** [comment]
Golfed version:
```
<Zoffix__> m: -> $ where /./, $/ {}("x", "y")
<camelia> rakudo-moar fffe89c5e: OUTPUT: «Cannot call method 'STORE' on a null object␤  in block <unit> at <tmp> line 1␤␤»
```

It's basically the regex in `where` trying to write to `$/`, but `$/` is used by a param and at that point it's null.

**Comment by vrurg** [comment]
Fixed with b777d6ff61e52232a784724ec5b6f855b5e50813

**Comment by lizmat** [comment]
Tests added with https://github.com/perl6/roast/commit/38dbffdb91
